packages/plugins/rubygem-smart-proxy-probing/rubygem-smart-proxy-probing.spec

```diff
@@ -0,0 +1,106 @@
+# Generated from smart-proxy-probing-0.0.2.gem by gem2rpm -*- rpm-spec -*-
+# template: smart_proxy_plugin
+%global gem_name smart-proxy-probing
+%global plugin_name smart-proxy-probing
+
+%global foreman_proxy_min_version 1.17
+%global foreman_proxy_dir %{_datarootdir}/foreman-proxy
+%global foreman_proxy_bundlerd_dir %{foreman_proxy_dir}/bundler.d
+%global foreman_proxy_settingsd_dir %{_sysconfdir}/foreman-proxy/settings.d
+%global smart_proxy_dynflow_bundlerd_dir %{?rhel:/opt/theforeman/tfm/root/}%{_datadir}/smart_proxy_dynflow_core/bundler.d
+
+Name: rubygem-%{gem_name}
+Version: 0.0.2
+Release: 1%{?foremandist}%{?dist}
+Summary: Gem to allow probing through smart-proxy
+Group: Applications/Internet
+License: GPLv3
+URL: https://github.com/adamruzicka/smart-proxy-probing
+Source0: https://rubygems.org/gems/%{gem_name}-%{version}.gem
+
+# start generated dependencies
+Requires: foreman-proxy >= %{foreman_proxy_min_version}
+Requires: ruby(release)
+Requires: ruby
+Requires: ruby(rubygems)
+Requires: rubygem(smart_proxy_dynflow) < 0.3.0
+Requires: rubygem(smart_proxy_dynflow) >= 0.1.0
+BuildRequires: ruby(release)
+BuildRequires: ruby
+BuildRequires: rubygems-devel
+BuildArch: noarch
+Provides: rubygem(%{gem_name}) = %{version}
+Provides: foreman-proxy-plugin-%{plugin_name}
+# end generated dependencies
+
+Requires: %{?scl_prefix}rubygem(foreman_probing_core)
+
+%description
+Gem to allow probing through smart-proxy.
+
+
+%package doc
+Summary: Documentation for %{name}
+Group: Documentation
+Requires: %{name} = %{version}-%{release}
+BuildArch: noarch
+
+%description doc
+Documentation for %{name}.
+
+%prep
+gem unpack %{SOURCE0}
+
+%setup -q -D -T -n  %{gem_name}-%{version}
+
+gem spec %{SOURCE0} -l --ruby > %{gem_name}.gemspec
+
+%build
+# Create the gem as gem install only works on a gem file
+gem build %{gem_name}.gemspec
+
+# %%gem_install compiles any C extensions and installs the gem into ./%%gem_dir
+# by default, so that we can move it into the buildroot in %%install
+%gem_install
+
+%install
+mkdir -p %{buildroot}%{gem_dir}
+cp -a .%{gem_dir}/* \
+        %{buildroot}%{gem_dir}/
+
+# bundler file
+mkdir -p %{buildroot}%{foreman_proxy_bundlerd_dir}
+cat <<EOF > %{buildroot}%{foreman_proxy_bundlerd_dir}/%{plugin_name}.rb
+gem 'smart-proxy-probing'
+EOF
+
+mkdir -p %{buildroot}%{smart_proxy_dynflow_bundlerd_dir}
+cat <<EOF > %{buildroot}%{smart_proxy_dynflow_bundlerd_dir}/foreman_probing_core.rb
+gem 'foreman_probing_core'
```

Comment:
It doesn't have a requires on this gem package so I suspect this will lead to loading issues.

Reply:
I rebased and test builded it with mock, the resulting rpm has got the requirement `rubygem(foreman_probing_core)` by autorequires.

```
$ rpm -qp --requires /var/lib/mock/epel-7-x86_64/result/rubygem-smart-proxy-probing-0.0.2-1.el7.noarch.rpm 
config(rubygem-smart-proxy-probing) = 0.0.2-1.el7
foreman-proxy >= 1.17
rpmlib(CompressedFileNames) <= 3.0.4-1
rpmlib(FileDigests) <= 4.6.0-1
rpmlib(PayloadFilesHavePrefix) <= 4.0-1
ruby
ruby(release)
ruby(rubygems)
rubygem(foreman_probing_core)
rubygem(smart_proxy_dynflow) >= 0.1.0
rubygem(smart_proxy_dynflow) < 0.3.0
rpmlib(PayloadIsXz) <= 5.2-1
```

Reply:
Also, there is an explicit entry now: https://github.com/theforeman/foreman-packaging/pull/2777/files#diff-dee879c987c774f55075a8af185215d3R36